common/models/templates/src/tokenization.ts

```diff
@@ -33,6 +33,42 @@ namespace models {
     let leftSpans  = wordBreaker(context.left || '') || [];
     let rightSpans = wordBreaker(context.right || '') || [];
 
+    let leftTail: Span;
+    if(leftSpans.length > 0) {
+      leftTail = leftSpans[leftSpans.length - 1];
+    }
+
+    // Handle any directly-pre-caret 'detokenization' cases - where for this _specific_ context,
+    // we should not make a token division where one normally would exist otherwise.
+    // Refer to https://github.com/keymanapp/keyman/issues/6572.
```

Comment:
'detokenization' is not very clear to me.

```suggestion
    // For the default wordbreaker, we need to special-case handling of single quote
    // because it is treated as a word-breaking character, despite it being commonly 
    // usded within words in many Latin-script languages.
    // 
    // Handle any directly-pre-caret 'detokenization' cases - where for this _specific_ context,
    // we should not make a token division where one normally would exist otherwise.
    // See #6572.
```